src/script_number.js

```diff
@@ -0,0 +1,66 @@
+function decode (buffer, maxLength, minimal) {
+  maxLength = maxLength || 4
+  minimal = minimal === undefined ? true : minimal
+
+  var length = buffer.length
+  if (length === 0) return 0
+  if (length > maxLength) throw new TypeError('Script number overflow')
+  if (minimal) {
+    if ((buffer[length - 1] & 0x7f) === 0) {
+      if (length <= 1 || (buffer[length - 2] & 0x80) === 0) throw new Error('Non-minimally encoded script number')
+    }
+  }
+
+  var result
+
+  // 40-bit
+  if (length === 5) {
+    var a = buffer.readUInt32LE(0)
+    var b = buffer.readUInt8(4)
+
+    if (b & 0x80) return -((b & ~0x80) * 0x100000000 + a)
+    return b * 0x100000000 + a
+  }
+
+  // 32-bit / 24-bit / 16-bit / 8-bit
+  for (var i = 0; i < length; ++i) {
+    result |= buffer[i] << (8 * i)
+  }
+
+  if (buffer[length - 1] & 0x80) return -(result & ~(0x80 << (8 * (length - 1))))
```

Comment:
@fanatid got the last statement with:

``` javascript
  var b = buffer[length - 1]
  if (b & 0x80) {
    result -= b * Math.pow(2, 8 * (length - 1))

    return -((b & ~0x80) * Math.pow(2, 8 * (length - 1)) + result)
  }
```

Not really worth it though, the readability is marginal and the performance is almost 50% worse.
With the current code, the difference between when bit-wise operators are OK and the 40-bit special case is explicit.